Cabal/src/Distribution/Simple/Build.hs

```diff
@@ -614,8 +614,11 @@ generateCode
   -> Verbosity
   -> IO (SymbolicPath Pkg (Dir Source), [ModuleName.ModuleName])
 generateCode codeGens nm pdesc bi lbi clbi verbosity = do
+  debug verbosity $ "generateCode: " <> prettyShow (package pdesc)
```

Comment:
There are a *lot* of extra `debug` statements here to help me identify where we were losing time. I can remove those if desired.

Reply:
Please can you remove these, I think you can see in the cost centre profile there is a hot-spot in `findCwdWithExtension`, which is also what you identified.